F50/Platform/Platform.android.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Android.App;
+using Android.Content;
+using Android.OS;
+
+namespace Xamarin.F50
+{
+	public partial class Platform
+	{
+		static Application application;
+		static ActivityLifecycleContextListener lifecycleListener;
+
+		internal static Context CurrentContext =>
+			lifecycleListener?.Context ?? application?.ApplicationContext;
+
+		internal static Activity CurrentActivity =>
+			lifecycleListener?.Activity;
+
+		public static void Init(Activity activity, Bundle bundle)
+		{
+			lifecycleListener = new ActivityLifecycleContextListener();
+			application = activity.Application;
+			application.RegisterActivityLifecycleCallbacks(lifecycleListener);
+		}
+	}
+
+	class ActivityLifecycleContextListener : Java.Lang.Object, Application.IActivityLifecycleCallbacks
+	{
+		Activity currentActivity = null;
+
+		public Context Context =>
+			currentActivity ?? Application.Context;
+
+		public Activity Activity =>
+			currentActivity;
+
+		public void OnActivityCreated(Activity activity, Bundle savedInstanceState)
+		{
+		}
+
+		public void OnActivityDestroyed(Activity activity)
+		{
+		}
+
+		public void OnActivityPaused(Activity activity)
+		{
+		}
+
+		public void OnActivityResumed(Activity activity)
+		{
+			currentActivity = activity;

```

Comment:
should this happen on create as well? What about when it is paused/closed should we null out? Multiple activities at the same time?

Reply:
So I don't think on create. I think onresume is all you want to ensure you have the foremost activity always. 

We may need to null it out on pause though....

There shouldn't be multiple activities hitting onresume? Can you think of a way to actually do that?

Reply:
I don't think so......
Android 7 multi-window stuff... idk not sure anyone uses it, but one would always be active I assume